src/aegis_ai_web/src/endpoints/kpi.py

```diff
@@ -0,0 +1,141 @@
+"""
+KPI endpoint module for CVE analysis feedback.
+"""
+
+import csv
+import fcntl
+import os
+from datetime import datetime
+from pathlib import Path
+from typing import List, Dict, Any
+
+from enum import Enum
+
+from fastapi import HTTPException
+
+from aegis_ai import get_settings
+from aegis_ai_web.src.data_models import FEEDBACK_SCHEMA, KPIScoreResponse
+
+
+def read_feedback_logs() -> List[Dict[str, str]]:
+    """
+    Read and parse feedback log entries from CSV file.
+
+    Returns:
+        List of Dict entries where all values are strings from CSV.
+        Returns empty list if file doesn't exist or has no valid entries.
+    """
+    # Read env var dynamically to support test fixtures that set it
+    log_file = os.getenv(
+        "AEGIS_WEB_FEEDBACK_LOG", f"{get_settings().config_dir}/feedback.csv"
+    )
+    log_path = Path(log_file)
+
+    # Return empty list if file doesn't exist
+    if not log_path.exists():
```

Comment:
This introduces a [TOCTOU](https://cwe.mitre.org/data/definitions/367.html) race, which will provoke SAST scanners.  While it is unlikely that the file would be deleted meanwhile, opening the file unconditionally and handling an exception in case it did not exist is IMO the saner approach here.